src/execution_plan/execution_plan_modify.c

```diff
@@ -203,26 +203,24 @@ OpBase *ExecutionPlan_LocateLastOp(OpBase *root, OPType type) {
 	return _ExecutionPlan_LocateOp(root, type, RTL);
 }
 
-OpBase *ExecutionPlan_LocateReferences(OpBase *root, const OpBase *recurse_limit,
-									   rax *refs_to_resolve) {
+static OpBase *_ExecutionPlan_LocateReferences(OpBase *root, const OpBase *recurse_limit,
+											   rax *refs_to_resolve) {
 	if(root == recurse_limit) return NULL; // Don't traverse into earlier ExecutionPlan scopes.
 
 	int dependency_count = 0;
 	OpBase *resolving_op = NULL;
 	bool all_refs_resolved = false;
 	for(int i = 0; i < root->childCount && !all_refs_resolved; i++) {
 		// Visit each child and try to resolve references, storing a pointer to the child if successful.
-		resolving_op = ExecutionPlan_LocateReferences(root->children[i], recurse_limit, refs_to_resolve);
-		if(resolving_op) dependency_count ++; // Count how many children resolved references.
+		OpBase *tmp_op = _ExecutionPlan_LocateReferences(root->children[i], recurse_limit, refs_to_resolve);
```

Comment:
Can you please explain this addition?

Reply:
when `resolving_op` was set with a value, and then another call to `ExecutionPlan_LocateReferences` returned `NULL`, then the previous version of the function set `resolving_op` to `NULL`